src/System.Linq/src/System/Linq/Where.cs

```diff
@@ -320,6 +458,8 @@ public override Iterator<TResult> Clone()
                 return new WhereSelectListIterator<TSource, TResult>(_source, _predicate, _selector);
             }
 
+            public int GetCount(bool onlyIfCheap) => onlyIfCheap ? -1 : EnumerableHelpers.Count(this);
```

Comment:
What's the reason for calling Count rather than open coding this, which would avoid the cost of invoking the selector for every element that passes the predicate?  Are we concerned about not throwing exceptions where we previously threw them?